sklearn/utils/estimator_checks.py

```diff
@@ -252,6 +241,21 @@ def _check_transformer(name, Transformer, X, y):
             assert_raises(ValueError, transformer.transform, X.T)
 
 
+def check_fit_score_takes_y(name, Estimator):
+    # check that all estimators accept an optional y
+    # in fit and score so they can be used in pipelines
+    rnd = np.random.RandomState(0)
+    X = rnd.uniform(size=(10, 3))
+    y = (X[:, 0] * 4).astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    with warnings.catch_warnings(record=True):
+        estimator = Estimator()
+    set_fast_parameters(estimator)
+    set_random_state(estimator)
+    estimator.fit(X, y)
```

Comment:
I think the second positional argument should always be ``y``. This is the case for all estimators but MDS #4132. At least in master. is the ``sample_weights`` before ``y`` in the clustering algorithms?
That would complicate things, and also break the current pipelining code.